### PR TITLE
Add function to create ICS 23 proof with tests

### PR DESCRIPTION
**Pull request by saideng**
## Description

Closes: #XXXX

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: store/verkle/store.go [hunk 1]
```diff
@@ -78,16 +78,54 @@ func LoadStore(db dbm.DBReadWriter) *Store {
...
+	return rootCommitment[:]
+}
+
 func (s *Store) GetRoot() verkle.VerkleNode {
 	return s.tree
 }
 
-func (s *Store) GetProof(_ []byte) (*tmcrypto.ProofOps, error) {
-	panic("not implemented")
+// GetTreeKV returns map of hash(kay) -> hash(value)
```
Comment (supragya) [inline store/verkle/store.go:90]: Typo

File: store/verkle/ics23.go [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+	kvs := store.GetTreeKV()
+	proof, _, _, _ := verkle.MakeVerkleMultiProof(store.tree, [][]byte{keyPath[:]}, kvs)
+	proofStr, _, err := verkle.SerializeProof(proof)
+	if err != nil {
+		return nil, err
+	}
+
+	var keys, vals [][]byte
+	for k, v := range kvs {
+		keys = append(keys, []byte(k))
```
Comment (supragya) [inline store/verkle/ics23.go:25]: Put a marking comment here: This is just for trials till go-verkle supports `proofKV`

File: store/verkle/ics23.go [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+	"github.com/gballet/go-verkle"
+	"golang.org/x/crypto/sha3"
+)
+
+func createIcs23Proof(store *Store, key []byte) (*ics23.CommitmentProof, error) {
+	ret := &ics23.CommitmentProof{}
+	keyPath := sha3.Sum256(key)
+	// TODO: should not use all KVs
+	kvs := store.GetTreeKV()
+	proof, _, _, _ := verkle.MakeVerkleMultiProof(store.tree, [][]byte{keyPath[:]}, kvs)
```
Comment (supragya) [inline store/verkle/ics23.go:17]: Does this contain within it the key which is included in proof. Should be, just verifying once.

**Review by supragya — CHANGES_REQUESTED** [review]

File: store/verkle/ics23.go (hunk 2 above)
Comment (saideng) [reply store/verkle/ics23.go:25]: Already have a comment at line 15.  Added a comment to this line as well.

**Review by supragya — APPROVED** [review]

**Pull request merged**